sql/2020/07_SEO/10_02_lang_attribute_raw.sql

```diff
@@ -0,0 +1,18 @@
+#standardSQL
+# 10_02: lang attribute usage and mistakes (lang='en')
+# source: https://discuss.httparchive.org/t/what-are-the-invalid-uses-of-the-lang-attribute/1022
+SELECT
+  client,
+  LOWER(REGEXP_EXTRACT(body, '(?i)<html[^>]*lang=[\'"]?([a-z]{2})')) AS lang,
+  COUNT(0) AS freq,
+  SUM(COUNT(0)) OVER (PARTITION BY client) AS total,
+  ROUND(COUNT(0) * 100 / SUM(COUNT(0)) OVER (PARTITION BY client), 2) AS pct
+FROM
+  `httparchive.almanac.summary_response_bodies`
```

Comment:
Querying the `response_bodies` tables is a red flag that this should be implemented as a custom metric instead.

I'd propose a new custom metric:

```js
document.querySelector('html')?.getAttribute('lang')
```

Reply:
Yep, I'll review all queries and see what I can move over.

Today I'm going to play and see if I can create a single summary_response_bodies query that can get everything we want in one go. The team would love for me to compare raw with rendered for certain critical bits of data that can cause crawling issues.

Reply:
I've moved a lot to custom metrics. Should I do a pull request from our forked almanac.js now so you can see where we are at?

Reply:
Thatd be great

Reply:
Just done it